app/stores/AppStore.js

```diff
@@ -51,18 +52,21 @@ export default class AppStore extends Store {
   }
 
   _redirectToLanguageSelectionIfNoLocaleSet = () => {
-    if (this.hasLoadedCurrentLocale && !this.isCurrentLocaleSet) {
+    const { isConnected } = this.stores.networkStatus;
+    if (isConnected && this.hasLoadedCurrentLocale && !this.isCurrentLocaleSet) {
       this.actions.router.goToRoute({ route: '/profile/language-selection' });
     }
   };
 
-  @action _updateLocale = ({ locale }: { locale: string }) => {
-    this.setProfileLocaleRequest.execute(locale).then(() => (
-      this.getProfileLocaleRequest.invalidate().patch(() => locale)
-    ))
-    .catch(action((error) => {
-      this.error = error;
-    }));
+  _redirectToLoadingScreenWhenDisconnected = () => {
+    if (!this.stores.networkStatus.isConnected) {
```

Comment:
I have introduced another reaction @tomothespian to cover the case when we are not client is not connected to node or connection is lost after language selection screen is displayed.